src/spec.c

```diff
@@ -539,18 +539,21 @@ static int parseVectorField(FieldSpec *fs, ArgsCursor *ac, QueryError *status) {
     fs->vecSimParams.algo = VecSimAlgo_BF;
     fs->vecSimParams.bfParams.initialCapacity = 1000;
     fs->vecSimParams.bfParams.blockSize = BF_DEFAULT_BLOCK_SIZE;
-    return parseVectorField_flat(fs, ac, status);
+    rc = parseVectorField_flat(fs, ac, status);
+    fs->expBlobSize = fs->vecSimParams.bfParams.dim * VecSimType_sizeof(fs->vecSimParams.bfParams.type);
   } else if (!strncasecmp(VECSIM_ALGORITHM_HNSW, algStr, len)) {
     fs->vecSimParams.algo = VecSimAlgo_HNSWLIB;
     fs->vecSimParams.hnswParams.initialCapacity = 1000;
     fs->vecSimParams.hnswParams.M = HNSW_DEFAULT_M;
     fs->vecSimParams.hnswParams.efConstruction = HNSW_DEFAULT_EF_C;
     fs->vecSimParams.hnswParams.efRuntime = HNSW_DEFAULT_EF_RT;
-    return parseVectorField_hnsw(fs, ac, status);
+    rc = parseVectorField_hnsw(fs, ac, status);
+    fs->expBlobSize = fs->vecSimParams.hnswParams.dim * VecSimType_sizeof(fs->vecSimParams.hnswParams.type);
```

Comment:
please have it on the parse function, and set it only upon successful parse, so if the parse fails you will not set/read invalid data